src/effects/VST3/VST3Effect.cpp

```diff
@@ -199,46 +195,32 @@ VST3Effect::VST3Effect(
 
    const auto& pluginFactory = mModule->getFactory();
    auto editController = FUnknownPtr<Vst::IEditController>(mEffectComponent);
-   if(!editController)
+   if(editController.get() == nullptr)
    {
-      //Plugin does not have a separate edit controller
       TUID controllerCID;
             
 		if (mEffectComponent->getControllerClassId (controllerCID) == kResultTrue)
-		{
-			if(mEditController = pluginFactory.createInstance<Vst::IEditController> (VST3::UID (controllerCID)))
-			{
-            if(mEditController->initialize(&LocalContext()) != kResultOk)
-            {
-               wxLogMessage("Cannot create edit controller: initialization failed");
-               mEditController = nullptr;
-            }
-			}
-         else
-            wxLogMessage("Cannot create edit controller: attempt failed");
-		}
+         editController = pluginFactory.createInstance<Vst::IEditController>(VST3::UID(controllerCID));
       else
-         wxLogMessage("Cannot create edit controller: failed read controller Class ID from the effect component");
+         throw std::runtime_error("Failed to instantiate edit controller");
    }
-   else
-      mEditController = editController;
+   
+   mEditController = editController;
+   mEditController->initialize(&AudacityVst3HostApplication::Get());
```

Comment:
There is now an assumption that editController is not null and initialize succeeds.  Why is it allowable now to remove the checks that the old version had?